### ISSUE TITLE
[Bug]: can't run validate on a stand-alone file

### ISSUE THREAD

**Issue by CodyCBakerPhD** [dandi/dandi-cli#1211]
This and some related issues came up while I was helping a user upload their data to DANDI earlier today.

I'm used to being able to run `dandi validate some_nwbfile.nwb` and have it tell me if it's valid or not. In fact, the [documentation on the archive](https://www.dandiarchive.org/handbook/13_upload/#data-uploadmanagement-workflow) still suggest this as the workflow prior to doing anything else.

Yet I know get an error using `dandi==0.49.0`

```
dandi validate test_upload.nwb
[DANDI.NO_DANDISET_FOUND] test_upload.nwb — File is not inside a Dandiset
```

I feel like the `DANDI.NO_DANDISET_FOUND` check should not be run when the file is not yet inside the dandiset.

Either that or someone needs to update the documentation for everything related to this.

**Comment by yarikoptic** [comment]
indeed, it became impossible after we added validation of the paths to correspond to either BIDS or DANDI layout. And indeed it might be useful to still allow validation of loose .nwb files before they are organized into a DANDI or BIDS dandiset. But we would need to issue a warning that validation is not full.

We should first finish up/merge #1203 and then approach this.

**Comment by yarikoptic** [comment]
thinking about it -- I think we still should fail, just informatively that the file(s) are not found belonging to any dandiset. Later we might add an option to be able to filterout/ignore some of the types of error.

**Comment by CodyCBakerPhD** [comment]
The main point of this issue then is how these changes to the CLI/API broke the documentation: https://www.dandiarchive.org/handbook/13_upload/#data-uploadmanagement-workflow

**Comment by TheChymera** [comment]
If this is just DANDI-less nwb validation, why should it be run via `dandi`?

**Comment by CodyCBakerPhD** [comment]
Because that's what the documentation has informed people to do for the nearly the past 3 years: https://github.com/dandi/handbook/blob/01a66f5f6ac2e00a8d1b6a7140802d55c0617948/docs/10_using_dandi.md#uploading-a-dandiset and still does to this day https://www.dandiarchive.org/handbook/13_upload/#data-uploadmanagement-workflow

So it's been understood as a core feature for all that time - kind of odd to have it removed all of a sudden without warning or updating the usage docs

Also, as I understood a part of the reason is to ensure that `dandi organize` actually has enough of the right metadata in the NWB files to give them unique names

**Comment by yarikoptic** [comment]
We will need to
- add an option to ignore that error
- update documentation

**Comment by yarikoptic** [comment]
```shell
❯ python3 -c 'import this' | grep -e Zen -e Ex
The Zen of Python, by Tim Peters
Explicit is better than implicit.
```
so I would consider 1a a little too implicit since then we can miss the case where people do consider running in a valid dandiset whenever they e.g. forgot to download `dandiset.yml`.

**Comment by yarikoptic** [comment]
I will consider this one resolved by #1234 and https://github.com/dandi/handbook/pull/79
